src/sage/combinat/root_system/root_system.py

```diff
@@ -787,6 +787,56 @@ def coambient_space(self, base_ring=QQ):
         """
         return self.dual.ambient_space(base_ring)
 
+    def coxeter_number(self):
+        """
+        For an irreducible finite root system, reports the Coxeter number.
+        This function is a wrapper for the corresponding function in CartanType.
+
+        EXAMPLES::
+
+            sage: rt = RootSystem(['C', 5])
+            sage: rt.coxeter_number()
+            10
+
+        .. SEEALSO:: :meth:`~sage.combinat.root_system.cartan_type.CartanType_standard_finite.coxeter_number`
+        """
+        # Check if RootSystem is finite and irreducible
+        if self.is_finite() and self.is_irreducible():
+            # Hand over to CartanType method
+            return self._cartan_type.coxeter_number()
+        else:
+            raise ValueError("Coxeter number is defined only for finite and irreducible root systems.")
```

Comment:
```suggestion
        if not (self.is_finite() and self.is_irreducible()):
            raise ValueError("the Coxeter number is defined only for finite and irreducible root systems")
        # Hand over to CartanType method
        return self._cartan_type.coxeter_number()
```
There isn't a need to give implementation details in the doc. I find this style of code a bit easier to understand and maintain.